src/helpers/figures/charts/scorecard_chart_config_builder.ts

```diff
@@ -218,27 +249,54 @@ class ScorecardChartConfigBuilder {
 
   private getTextStyles() {
     // If the widest text overflows horizontally, scale it down, and apply the same scaling factors to all the other fonts.
-    const maxLineWidth = this.width * (1 - 2 * CHART_PADDING_RATIO);
-    const widestElement = this.getWidestElement();
-    const baseFontSize = widestElement.getElementMaxFontSize(this.getDrawableHeight(), this);
-    const fontSizeMatchingWidth = getFontSizeMatchingWidth(
-      maxLineWidth,
-      baseFontSize,
-      (fontSize: number) => widestElement.getElementWidth(fontSize, this.context, this)
-    );
-    let scalingFactor = fontSizeMatchingWidth / baseFontSize;
+    const maxLineWidth = this.width - 2 * CHART_PADDING;
+    const drawableHeight = this.getDrawableHeight;
 
     // Fonts sizes in px
-    const keyFontSize =
-      new KeyValueElement(this.runtime.keyValueStyle).getElementMaxFontSize(
-        this.getDrawableHeight(),
-        this
-      ) * scalingFactor;
-    const baselineFontSize =
-      new BaselineElement(this.runtime.baselineStyle).getElementMaxFontSize(
-        this.getDrawableHeight(),
-        this
-      ) * scalingFactor;
+    const keyValueElement = new KeyValueElement(this.runtime.keyValueStyle);
+    const heightFont = keyValueElement.getElementMaxFontSize(drawableHeight, this);
+    const widthFont = getFontSizeMatchingWidth(maxLineWidth, 600, (fontSize: number) =>
+      keyValueElement.getElementWidth(fontSize, this.context, this)
+    );
+    const keyFontSize = Math.min(heightFont, widthFont);
+    let baselineValueFontSize = Math.floor(keyFontSize * 0.5);
+
+    this.context.font = getDefaultContextFont(
+      baselineValueFontSize,
+      this.runtime.baselineStyle?.bold,
+      this.runtime.baselineStyle?.italic
+    );
+    const baselineText = this.baselineArrow !== "neutral" ? "A " + this.baseline : this.baseline;
+    const baselineValueMeasure = this.context.measureText(baselineText);
+    const remainingWidth = maxLineWidth - baselineValueMeasure.width;
+    let baselineDescrFontSize = getFontSizeMatchingWidth(
+      remainingWidth,
+      baselineValueFontSize,
+      (fontSize: number) => {
+        this.context.font = getDefaultContextFont(fontSize);
+        return this.context.measureText(this.baselineDescr).width;
+      }
+    );
+
+    let splittedBaseline = false;
+    if (baselineDescrFontSize < baselineValueFontSize / 2.5) {
+      splittedBaseline = true;
+      baselineDescrFontSize = Math.floor(baselineValueFontSize / 2.5);
+      for (const str of splitText(this.baselineDescr)) {
```

Comment:
We should probably put the splitted baseline in the `ScorecardChartConfig`,rather than hope that we split the exact same way in both this function and the text rendering function